lib/active_campaign_crm/connection.rb

```diff
@@ -16,30 +17,60 @@ def get(url, params = {})
         req.headers['Api-Token'] = ActiveCampaignCrm.configuration.api_key
         req.params = params
       end
-      JSON.parse response.body
+      handle_response(response)
     end
 
     def post(url, body)
-      @connection.post do |req|
+      response = @connection.post do |req|
         req.url url
         req.headers['Api-Token'] = ActiveCampaignCrm.configuration.api_key
         req.body = body
       end
+      handle_response(response)
     end
 
     def put(url, body)
-      @connection.put do |req|
+      response = @connection.put do |req|
         req.url url
         req.headers['Api-Token'] = ActiveCampaignCrm.configuration.api_key
         req.body = body
       end
+      handle_response(response)
     end
 
     def delete(url)
-      @connection.delete do |req|
+      response = @connection.delete do |req|
         req.url url
         req.headers['Api-Token'] = ActiveCampaignCrm.configuration.api_key
       end
+      handle_response(response)
+    end
+
+    def handle_response(response)
+      return JSON.parse response.body if response.success?
```

Comment:
are you using a linter for this? if you're not you should. I'd prefer to wrap `response-body` in `()`

Reply:
changed